### DISCUSSION TITLE
Getting different ratios between program and test code

### DISCUSSION THREAD

**Discussion by Matt-Roden** [rapidfuzz/RapidFuzz#222] [Q&A]
I am using rapidFuzz as a means to give a search bar some fuzzy matching logic. While testing my code, I noticed that I am getting different ratio values between the tests and my actual program. 

For example, I have now set my python code up to print tuples containing the two strings that are being compared, as well as their ratio value. Meaning, when I make a search in the browser from the client, I am printing these values to my terminal. And when I run my test suite against this feature, it also prints these same values. The problem is, I am getting a difference of almost 10% between my client side making a fuzzy search and my test suite making a fuzzy search. Basically, my client side code is running great and the library is working as it should. But in my test suite, since the ratio value is 10% lower, I'm not getting back the true expected results, thus failing a test that _should_ be passing. 

Example: 
- User query's "Test Bendor 2" (the db contains _Test Vendor_ and _Test Vendor 2_)
- Expected Result of getting both _Test Vendor 2_ and _Test Vendor_ back work on client side when making request from localhost in the browser. 
- From the test suite though, the expected result is the same, except _Test Vendor_ is not returned anymore due to a lower ratio when the rapidFuzz tools are being utilized from the test. 

Any clue on why this could be happening? Is it a python or rapidFuzz thing?

**Comment by maxbachmann** [comment]
From your description rapidfuzz is called with the same input in both cases and it is called in exactly the same manner. In this case I would not expect any differences in the results either.

**Reply by Matt-Roden** [reply]
Exactly. It was a head scratcher for my team and I at work. Any how, we used a python mock instead to get passing tests that match the actual output we're getting in the program code. Could be something with the python interpreter? Very bizarre. Will report back if something else is found out. Thanks for the response!

**Reply by maxbachmann** [reply]
> Could be something with the python interpreter?

Nothing I am aware of at least with standard CPython (PyPy had a bug with string encoding in their C-API a while back, but that's fixed for quite a while already). In case you find out the cause / a way to reproduce this, I would definitely be interested.
